### PR TITLE
PLU-318: [TEMPLATES-4]: Add template subpage

### PR DESCRIPTION
**Pull request by m0nggh**
## Problem
Users are unaware of what they can do with Plumber.

## Solution
Introduce templates to speed up their pipe creation and provide new ideas for them to automate. 

## Details
- Add `getTemplate` graphQL query to fetch template or demo template based on the id, blocks the query if the id is invalid
- Add template subpage which is a modal
  - Any other template will have a normal step graphic (step 1, step 2...)
  - Check the route support enquiries for ifThen step graphic
- Add a `sampleUrlDescription` to each template step that is `formsg` or `tiles`, this opens up a URL to a form or tile for that template

## Tests
- [x] Can access each template by clicking on the tile and it should open a modal with the route `/template/:templateId`
- [x] Invalid template appears if wrong id is input e.g. `/template/123`

### CODE REVIEW

**Comment by m0nggh** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/717?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#717** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/717?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#748** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/748?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 13 other dependent PRs ([#714](https://github.com/opengovsg/plumber/pull/714) <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/714?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#715](https://github.com/opengovsg/plumber/pull/715) <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/715?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#716](https://github.com/opengovsg/plumber/pull/716) <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/716?utm_source=stack-comment-icon" target="_blank">Graphite</a> and 10 others)
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @m0nggh and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/src/pages/Template/components/TemplateBody.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      justifyContent="center"
+      alignItems="center"
+      w={{ base: '100%', md: '90%', lg: '60%' }}
+      mx="auto"
+      py={{ base: '0.75rem', md: '1.5rem' }}
+    >
+      <Infobox icon={<></>} variant="primary" mb={6}>
+        <Text textStyle="body-1">
+          This is a preview of your workflow. Add details to use this template
+          as is. Or, add more steps to customise it to your use case.
```
Comment (pregnantboy) [inline packages/frontend/src/pages/Template/components/TemplateBody.tsx]:
```suggestion
          as-is. Or, add more steps to customise it to your use case.
```

File: packages/frontend/src/pages/Template/InvalidTemplatePage.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Stack, Text } from '@chakra-ui/react'
+
+export default function InvalidTemplatePage() {
+  // TODO (mal): check if this needs to be beautified
+  return (
+    <Stack mt={24} justifyContent="center" alignItems="center" gap={2}>
+      <Text textStyle="subhead-1">
+        Template not found, please refer to our templates page!
```
Comment (pregnantboy) [inline packages/frontend/src/pages/Template/InvalidTemplatePage.tsx]:
```suggestion
        Template not found, please refer to our <link>templates</link> page!
```
maybe link back to templates page

Comment (m0nggh) [reply packages/frontend/src/pages/Template/InvalidTemplatePage.tsx]: im going to remove this page and only allow users to load the template modal if it is valid, so `templates/abc` will just remain on the same page

File: packages/frontend/src/pages/Template/components/TemplateBody.tsx (hunk 1 above)
Comment (m0nggh) [reply packages/frontend/src/pages/Template/components/TemplateBody.tsx]: wow TIL

File: packages/frontend/src/pages/Template/components/IfThenTemplateStepContent.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        borderTopRadius={8}
+        borderBottomRadius={0}
+      >
+        <Flex
+          borderWidth={1}
+          boxSize={16}
+          borderColor="base.divider.strong"
+          justifyContent="center"
+          alignItems="center"
+          borderRadius="0.25rem"
```
Comment (pregnantboy) [inline packages/frontend/src/pages/Template/components/IfThenTemplateStepContent.tsx]: prefer not use rem, but use one of chakra's default size 
https://v2.chakra-ui.com/docs/styled-system/theme#border-radius

File: packages/frontend/src/pages/Template/components/TemplateStepContent.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+const FALLBACK_EVENT_NAME = 'Sample Event'
+
+export default function TemplateStepContent(props: TemplateStepContentProps) {
+  const { app, templateStep } = props
+  const { eventKey, position, sampleUrl, sampleUrlDescription } = templateStep
+  // sanity check
+  if (!app) {
+    return <></>
+  }
+
```
Comment (pregnantboy) [inline packages/frontend/src/pages/Template/components/TemplateStepContent.tsx:22]: nit:
```
const isTrigger = position === 1
```

File: packages/frontend/src/pages/Templates/index.tsx [hunk 5]
```diff
@@ -10,71 +12,88 @@ import { Template } from '@/graphql/__generated__/graphql'
...
+
 const TEMPLATES_TITLE = 'Templates'
 
 export default function Templates(): JSX.Element {
+  const navigate = useNavigate()
   const { data, loading } = useQuery(GET_TEMPLATES)
+
   const templates: Template[] = data?.getTemplates
+  const { templateId } = useParams()
+  const template = useMemo(
```
Comment (pregnantboy) [inline packages/frontend/src/pages/Templates/index.tsx]: unnecessary use memo

File: packages/frontend/src/pages/Templates/index.tsx [hunk 6]
```diff
@@ -10,71 +12,88 @@ import { Template } from '@/graphql/__generated__/graphql'
...
-        <Text textStyle="body-1">
-          Pre-built pipes that you can use as is or customise further for your
-          own use case
-        </Text>
-      </Flex>
+    <>
+      <Container
+        py={9}
+        pl={{ base: '2rem', xl: '3.5rem' }}
+        pr={{ base: '2rem', xl: '8.5rem' }}
```
Comment (pregnantboy) [inline packages/frontend/src/pages/Templates/index.tsx:30]: to refactor later

**Review by pregnantboy — APPROVED** [review]
nits only

File: packages/frontend/src/pages/Template/components/TemplateStepContent.tsx (hunk 4 above)
Comment (m0nggh) [reply packages/frontend/src/pages/Template/components/TemplateStepContent.tsx:22]: thanks

**Pull request merged**